distributed/shuffle/shuffle_extension.py

```diff
@@ -333,6 +344,23 @@ def get_output_partition(
             )
         return output
 
+    @overload
+    async def _get_shuffle(
+        self,
+        shuffle_id: ShuffleId,
+    ) -> Shuffle:
+        ...
+
+    @overload
+    async def _get_shuffle(
+        self,
+        shuffle_id: ShuffleId,
+        empty: pd.DataFrame,
+        column: str,
+        npartitions: int,
+    ) -> Shuffle:
+        ...
```

Comment:
This kind of functions must either be called with all or nothing, hence the overload